WPFHospitalEditor/Service/MapObjectService.cs

```diff
@@ -38,5 +46,41 @@ private List<MapObject> colorSelectedObject(List<MapObject> allMapObjects)
             }
             return allMapObjects;
         }
+
+        public List<MapObject> checkMapObjectSearchInput(List<MapObject> mapObjects, string textBoxInput, string comboBoxTextInput)
+        {
+            List<MapObject> searchResult = new List<MapObject>();
+            MessageBox.Show("Lista: " + mapObjects.Count + " TextBox: " + textBoxInput + "ComboBox: " + comboBoxTextInput);
+            foreach (MapObject mapObject in mapObjects)
+            {
+                if (textBoxEmpty(mapObject, textBoxInput, comboBoxTextInput))
+                {
+                    searchResult.Add(mapObject);
+                }
+                else if (typeNotChosen(mapObject, textBoxInput, comboBoxTextInput))
+                {
+                    searchResult.Add(mapObject);
+                }
+                else if (bothParametersActive(mapObject, textBoxInput, comboBoxTextInput))
+                {
+                    searchResult.Add(mapObject);
+                }
```

Comment:
In all condition functions you send the same parameters and execute the same code if the outcome of the condition is true. You can create one function that will replace all three.

Reply:
Good point, it is all one method now.